src/indexer.cc

```diff
@@ -1611,6 +1616,36 @@ void OnIndexReference(CXClientData client_data, const CXIdxEntityRefInfo* ref) {
 FileContents::FileContents(const std::string& path, const std::string& content)
     : path(path), content(content) {}
 
+FileContentsWithOffsets::FileContentsWithOffsets() : line_offsets_{0} {}
+
+FileContentsWithOffsets::FileContentsWithOffsets(std::string s) {
+  contents_ = s;
+  line_offsets_.push_back(0);
+  for (size_t i = 0; i < s.size(); i++)
+    if (s[i] == '\r' || s[i] == '\n') {
```

Comment:
Do you even need to consider '\r'?

Reply:
I want to support all of '\r' '\n' "\r\n"

https://en.wikipedia.org/wiki/Newline#Representations_in_different_character_encoding_specifications

Maybe '\r' is no longer used.

Reply:
It doesn't seem to be used by any of the modern OSes.

Reply:
Kept only '\n' because "\r\n" can be handled by the same code.